### PR TITLE
[4414][FIX] stock_move_line_move_location

### PR DESCRIPTION
**Pull request by kanda999**
[4414](https://www.quartile.co/web#id=4414&menu_id=505&cids=3&action=1457&model=project.task&view_type=form)

Fix there is only one transfer line proposed when I seleccted two stock move lines. 
2024-04-09_12h06_58
(one stock move line with owner and one stock move line without owner)

### CODE REVIEW

**Comment by kanda999** [comment]
@AungKoKoLin1997 
I guess you were referring to this part of the stock_move_line module.
https://github.com/qrtl/axls-oca/blob/2a1e6749abb68d187033c17bcc11fa7c4d6923b7/stock_move_location/wizard/stock_move_location.py#L116

Do we need to group stock_move_line as well? 
I think no need for that.

**Comment by AungKoKoLin1997** [comment]
> Do we need to group stock_move_line as well?
> I think no need for that.

@kanda999 Yes. It is needed and I did for intentionally. Why should we show multiple lines in wizard for the lines that have same product_id and lot_id?

**Comment by AungKoKoLin1997** [comment]
As per this fix, you should not remove groups part and should add owner_id in grouping parameters.

File: stock_move_line_move_location/wizard/stock_move_location.py [hunk 1]
```diff
@@ -37,15 +37,17 @@ def _prepare_wizard_move_lines(self, move_lines):
             return super()._prepare_wizard_move_lines(move_lines)
         # if need move only available qty per product on location
         for _product, ml in groupby(
-            sorted(move_lines, key=lambda r: (r.product_id, r.lot_id)),
-            lambda r: (r.product_id, r.lot_id),
+            sorted(move_lines, key=lambda r: (r.product_id, r.lot_id, r.owner_id)),
+            lambda r: (r.product_id, r.lot_id, r.owner_id),
```
Comment (yostashiro) [inline stock_move_line_move_location/wizard/stock_move_location.py]: Any reason `r.package_id` is not included?

**Review by yostashiro — APPROVED** [review]
Code review.

**Comment by kanda999** [comment]
Functional review: LGTM

**Pull request merged**